provider/machinepool/machine_pool_resource.go

```diff
@@ -989,6 +999,11 @@ func (r *MachinePoolResource) populateState(object *cmv1.MachinePool, state *Mac
 	} else {
 		state.AvailabilityZone = types.StringValue("")
 	}
+	azListValue, err := common.StringArrayToList(azs)
+	if err != nil {
+		return err
+	}
+	state.AvailabilityZones = azListValue
```

Comment:
I think that this should be done only in case `len(azs) != 1`, to avoid having two confusing similar attributes